src/Microsoft.Performance.SDK/Processing/ICustomDataSource.cs

```diff
@@ -132,5 +132,22 @@ public interface ICustomDataSource
         ///     <c>false</c> otherwise.
         /// </returns>
         bool IsDataSourceSupported(IDataSource dataSource);
+
+        /// <summary>
+        ///     Provides a method for this Custom Data Source to do any
+        ///     special cleanup operations for processors created by
+        ///     this instance, if applicable.
+        ///     <para />
+        ///     It is guaranteed that the instance passed to this method
+        ///     was created by this instance via call to one of the following:
+        ///     <list type="bullet">
+        ///         <item><see cref="CreateProcessor(IDataSource, IProcessorEnvironment, ProcessorOptions)"/></item>
+        ///         <item><see cref="CreateProcessor(IEnumerable{IDataSource}, IProcessorEnvironment, ProcessorOptions)"/></item>
+        ///     </list>
```

Comment:
Is this necessary? "guaranteed" is strong language and if we add additional methods, it might be easy to miss updating this... (and then our guarantee is meaningless and the world will crumble into dust). maybe we should just leave off the method list?

Reply:
That's a good point, I'll remove the list.